ZetaPushSwift/ZetaPushMacroService.swift

```diff
@@ -72,307 +84,265 @@ public enum ZetaPushMacroError: Error {
   }
 }
 
+// MARK: - ZetaPushMacroService
 open class ZetaPushMacroService : NSObject {
+  // MARK: Properties
+  open var onMacroError : ZPMacroServiceErrorBlock?
+  
+  public var clientHelper: ClientHelper?
+  var deploymentId: String?
+  var macroChannel: String?
+  var macroChannelError: String?
+  var macroChannelTrace: String?
+  
+  var channelSubscriptionBlocks = Dictionary<String, Array<Subscription>>()
+  
+  let log = XCGLogger(identifier: "macroserviceLogger", includeDefaultDestinations: true)
+  
+  // Callback for /completed macro channel
+  lazy var channelBlockMacroCompleted:ChannelSubscriptionBlock = { messageDict -> Void in
+    self.log.debug("ZetaPushMacroService channelBlockMacroCompleted")
+    self.log.debug(messageDict)
     
-    open var onMacroError : ZPMacroServiceErrorBlock?
-    
-    public var clientHelper: ClientHelper?
-    var deploymentId: String?
-    var macroChannel: String?
-    var macroChannelError: String?
-    var macroChannelTrace: String?
+    let macroChannel = self.composeServiceChannel(messageDict["name"] as? String ?? "")
```

Comment:
here we need to handle this with a guard let to log the case of `messageDict["name"]` is nil, because of previously it was handled with a force unwrap.